### PR TITLE
map zooming and right click menu change

### PR DESCRIPTION
**Pull request by EthanVieira**
* allow zooming on map with mouse scroll
* place item name in RCM in the middle of remaining space

### CODE REVIEW

File: src/clickable-object.js [hunk 1]
```diff
@@ -24,11 +25,21 @@ export class ClickableObject {
...
         let halfHeight = this.RCM_HALF_HEIGHTS[numActions - 1];
 
+        let actionTextSize = 0;
+
+        actions.forEach((action) => {
+            actionTextSize = Math.max(
+                actionTextSize,
+                getEstimatedPixelLength(action.text)
+            );
+        });
```
Comment (CurtisGreen) [inline src/clickable-object.js]:
```js
const actionTextWidth = actions
    .map((action) => getEstimatedPixelLength(action.text))
    .reduce((widest, current) => Math.max(widest, current), 0);
```

File: src/utilities.js [hunk 2]
```diff
@@ -159,6 +159,33 @@ export function getItemText(amount) {
...
+                totalLength += 8;
+                break;
+            case str[i] == str[i].toUpperCase():
+                totalLength += 8;
+                break;
+            default:
+                totalLength += 6;
+                break;
+        }
+    }
```
Comment (CurtisGreen) [inline src/utilities.js]: Recently I've been liking the if return, if return... format over switch case in times like this where switch isn't actually used

```js
export const getEstimatedPixelWidth = (str) => 
    str.split("").reduce((sum, char) => {
        if (["i", "l"].includes(char)) return sum + 1;
        if (["m", "w"].includes(char)) return sum + 8;
        if (char == char.toUpperCase()) return sum + 8;
        return sum + 6;
    }, 0);
```

File: src/scenes/world-map.js [hunk 3]
```diff
@@ -201,6 +202,26 @@ export class WorldMapScene extends Phaser.Scene {
         this.events.on("shutdown", () => {
             this.scale.resize(SCREEN.WIDTH, SCREEN.HEIGHT);
         });
+
+        container.on("wheel", function (pointer, deltaX, deltaY, deltaZ) {
```
Comment (CurtisGreen) [inline src/scenes/world-map.js:206]: Overall this is working but there are some weird edge cases at the edge of the map. I think it's good enough right now but worth keeping in mind

Comment (EthanVieira) [reply src/scenes/world-map.js:206]: Agreed. I struggled with some weird behavior for awhile and not sure what is the best solution for what we have, but this works good enough for me

**Review by CurtisGreen — APPROVED** [review]

**Pull request merged**